netbox/core/management/commands/syncdatasource.py

```diff
@@ -34,13 +34,12 @@ def handle(self, *args, **options):
         for i, datasource in enumerate(datasources, start=1):
             self.stdout.write(f"[{i}] Syncing {datasource}... ", ending='')
             self.stdout.flush()
-            try:
-                datasource.sync()
-                self.stdout.write(datasource.get_status_display())
-                self.stdout.flush()
-            except Exception as e:
-                DataSource.objects.filter(pk=datasource.pk).update(status=DataSourceStatusChoices.FAILED)
-                raise e
+
+            datasource.enqueue_sync_job()
```

Comment:
IMO we should leave this as-is and avoid the use of a background job entirely. We want the management command to block on the call to `sync()`.

Reply:
When `enqueue_sync_job()` gets no request passed as argument, the job will be run in foreground and block until `sync()` has finished. See netbox/core/models/data.py line 169. Maybe a dedicated `background=False` parameter would make this more clear than just using some automation?

Reply:
Right, but there's no reason to enqueue a job at all; the management command intends to call `sync()` directly. This change can be reverted.